core/org.openjdk.jmc.flightrecorder.rules.jdk/src/main/java/org/openjdk/jmc/flightrecorder/rules/jdk/util/ItemResultSetFactory.java

```diff
@@ -40,6 +40,10 @@
  */
 public class ItemResultSetFactory {
 	public IItemResultSet createResultSet(IItemCollection items, IItemQuery query) {
-		return new DefaultIItemResultSet(items, query);
+		return new DefaultIItemResultSet(items, query, 0);
```

Comment:
The issue with a value of 0 means the termination will be effective immediately, I think you can choose either
* a guard value like `-1` is better and in `DefaultIItemResultSet` just skip the termination.
* chose a very big value like one day (that's what JDK 19 does in `ExecutorService.close`, but the code can even wait longer)